plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1033,8 +1043,8 @@ void producer_plugin::set_program_options(
           "account that can not access to extended CPU/NET virtual resources")
          ("greylist-limit", boost::program_options::value<uint32_t>()->default_value(1000),
           "Limit (between 1 and 1000) on the multiple that CPU/NET virtual resources can extend during low usage (only enforced subjectively; use 1000 to not enforce any limit)")
-         ("cpu-effort-percent", bpo::value<uint32_t>()->default_value(config::default_block_cpu_effort_pct / config::percent_1),
-          "Percentage of cpu block production time used to produce block. Whole number percentages, e.g. 80 for 80%")
+         ("produce-block-offset-ms", bpo::value<uint32_t>()->default_value(config::default_produce_block_offset_ms),
+          "The number of milliseconds early the last block of a production round should be produced.")
```

Comment:
really it defines the latest time in the round the last block will be produced. It could be produced much earlier than that.

Reply:
`The number of milliseconds early the last block of a production round can be produced.` ?
`The minimum time to allow for the last block of a production round to make it to next block producer.`
`The minimum time to produce the last block of a production round.

Reply:
`The minimum time to reserve at the end of a production round for blocks to propagate to the next block producer.`
`The minimum offset from the end of a block production round to reserve for block propagation.`

Reply:
`Set to a value large enough for your blocks to have time to reach the next block producer.`

Reply:
This variable defines the point in time `produce-block-offset-ms` before the end of the 12 block round, and nodeos will ensure that the last block of the round is produced by this time at the latest.

Reply:
Reading them all again, I think: `The minimum time to reserve at the end of a production round for blocks to propagate to the next block producer.`

Reply:
excellent!